gateway/src/apicast/policy/logging/logging.lua

```diff
@@ -29,12 +40,101 @@ function _M.new(config)
   end
 
   self.enable_access_logs_val = val_for_ngx_var[enable_access_logs]
+  self.custom_logging = config.custom_logging
+  self.enable_json_logs = config.enable_json_logs
+	self.json_object_config = config.json_object_config or {}
+
+  if config.condition then
+    ngx.log(ngx.DEBUG, 'Enabling extended log with conditions')
+    local operations = {}
+    for _, operation in ipairs(config.condition.operations) do
+      table.insert( operations,
+        Operation.new(
+          operation.match, operation.match_type,
+          operation.op,
+          operation.value, operation.value_type or default_template_type))
+    end
+    self.condition = Condition.new( operations, config.condition.combine_op or default_combine_op)
+  end
 
   return self
 end
 
-function _M:log()
+local function get_request_context(context)
```

Comment:
Is there a reason to define these headers here instead of reusing `ngx_variable` ?
I think everything on `ngx_variable` should be available. We should define there the response headers, though.

Reply:
just because it is easy to provide to users information about the request and response in an easy way.

Reply:
But this limits us to the information in the service plus the headers.
I guess it's fine for now but in the future, users might ask for other info that we have in `ngx_variable` like it happened for other policies.